tools/ci_build/build.py

```diff
@@ -1158,6 +1158,7 @@ def generate_build_tree(
                     f"Float 8 types require CUDA>=11.8. They must be disabled on CUDA=={args.cuda_version}. "
                     f"Add '--disable_types float8' to your command line. See option disable_types."
                 )
+        cmake_args.append(f"-DCMAKE_CUDA_COMPILER={cuda_home}/bin/nvcc")
```

Comment:
The var does not always exist.  For example, if someone builds ONNX Runtime on Windows and the user has installed CUDA SDK's VS extension, then this script could be invoked in two different ways:
```
--use_cuda --cuda_version 11.8
--use_cuda --cuda_home somedir
```

Reply:
It shall be fine. User has to either use --cuda_home or define an env variable CUDA_HOME.

There is a check of cuda_home:
https://github.com/microsoft/onnxruntime/blob/7b822c3f11a08d8a80d6b6157289d0e7e9d22092/tools/ci_build/build.py#L1745-L1751

Reply:
not is_windows()

Reply:
cudnn_home is optional for Windows, cuda_home shall be valid.

Reply:
It looks confusing. You are right, but it shouldn't be required.